core/arch/arm/kernel/kern.ld.S

```diff
@@ -329,6 +335,11 @@ SECTIONS
 		KEEP(*(.rodata.dtdrv))
 		__rodata_dtdrv_end = .;
 #endif
+#ifdef CFG_EARLY_TA
+		__rodata_early_ta_start = .;
```

Comment:
Good catch. Strictly speaking we should have `. = ALIGN(__alignof__(struct early_ta))` but that can't be understood by the linker. Our options are:
1. Hard-code 4, which I expect to be the value of `__alignof__(struct early_ta)` for pretty much all platforms currently.
2. Use 8 for greater safety (?)
3. Add a definition to asm-defines.c.

I am tempted to do (3) although it may look weird to use this "asm" file for that purpose. Thoughts?